lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -1379,6 +1379,14 @@ parseURI (T.unpack -> uri) = runIdentity $ runExceptT $ do
             _ -> Right uri'
     either (throwE . TextDecodingError) pure res
 
+parseFileURI :: Text -> Either TextDecodingError URI
+parseFileURI (T.unpack -> uriStr) = runIdentity $ runExceptT $ do
+    uri <- parseAbsoluteURI uriStr ??
+        (TextDecodingError "Not a valid absolute URI.")
+    when (uriScheme uri /= "file:") $ fail "Not a valid URI scheme."
+    when (isNothing $ uriAuthority uri) $ fail "URI must contain a domain part."
```

Comment:
Hmm. Why?

Reply:
It was present in the other parser, and I thought it would be worth keeping.

Reply:
Well, for a server URL it makes sense. For a local file, it doesn't make much sense IMO... although:

```hs
λ: Just uri = parseURI "file:///home/ktorz/metadata.json"
λ: uriAuthority uri
Just (URIAuth {uriUserInfo = "", uriRegName = "", uriPort = ""}) 
```

I am not quite sure when this leads to `Nothing` :thinking: ....